modules/javafx.graphics/src/main/java/com/sun/javafx/tk/Toolkit.java

```diff
@@ -594,23 +594,12 @@ private Object getPaint(RadialGradient paint) {
     }
 
     public Object getPaint(Paint paint) {
-        if (paint instanceof Color) {
-            return createColorPaint((Color) paint);
-        }
-
-        if (paint instanceof LinearGradient) {
-            return getPaint((LinearGradient) paint);
-        }
-
-        if (paint instanceof RadialGradient) {
-            return getPaint((RadialGradient) paint);
-        }
-
-        if (paint instanceof ImagePattern) {
-            return createImagePatternPaint((ImagePattern) paint);
-        }
-
-        return null;
+        return switch (paint) {
```

Comment:
standard warning: not equivalent change (in the case of null `paint`).
it's ok in this case, since it looks like this method is never called with a null argument thanks to guards like this replicated across the code base:

```
peer.setFillPaint(getFill() == null ? null : tk.getPaint(getFill()));
```

would it have been easier to allow null argument and return null from `getPaint()` instead?

Reply:
`null` is not a valid argument for this method, and has never been. `acc_getPlatformPaint()` previously asserted that the return value is not `null`.

Reply:
this is not what I am asking though: in more than one place we have _callers_ like this
```
peer.setFillPaint(getFill() == null ? null : tk.getPaint(getFill()));
```

it could be changed to 
```
peer.setFillPaint(tk.getPaint(getFill()));
```

Reply:
Yes, that would make the callsite a bit easier. However, from an API perspective, I think it's not the best approach to handle nulls in the toolkit layer. A system that channels nulls thorugh various layers tends to become harder to reason about over time.

Reply:
I disagree: it makes it more resilient by doing what the caller has to do anyway.
However, the answer is acceptable.